src/Sarif.Viewer.VisualStudio/CodeLocationObject.cs

```diff
@@ -195,6 +195,8 @@ public void NavigateTo(bool usePreviewPane = true)
 
         public void ApplyDefaultSourceFileHighlighting()
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
`ThreadHelper.ThrowIfNotOnUIThread()` is also present at least 2 frames down the call graph (and also one up). What is the purpose of the redundancy?

Reply:
The analyzers pop that this must be called. I would also prefer to not have the redundancy.

---
In reply to: [499329815](https://github.com/microsoft/sarif-visualstudio-extension/pull/206#discussion_r499329815) [](ancestors = 499329815)